src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -92,7 +95,10 @@ public function refreshUser(UserInterface $user)
 
             $refreshedUser = $repository->find($id);
             if (null === $refreshedUser) {
-                throw new UsernameNotFoundException('User with id '.json_encode($id).' not found.');
+                $ex = new UsernameNotFoundException('User with id '.json_encode($id).' not found.');
+                $ex->setUsername(json_encode($id));
```

Comment:
why json_encoding here?

Reply:
See for example

https://github.com/symfony/symfony/blob/91d8bebfaca159bcbd0ab9bde78ca1790474f73b/src/Symfony/Bridge/Doctrine/Tests/Security/User/EntityUserProviderTest.php#L129

Here `$id` is not a string or even string like, while the username field is documented to be a string. I guess `UsernameNotFoundException` does not convey what the exact error is, because `$id` is not a username.